backend/src/v4/handler/alluxio.js

```diff
@@ -0,0 +1,90 @@
+/**
+ *  Copyright (C) 2020 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+"use strict";
+
+const config = require("../config.js");
+const path = require("path");
+const ResponseCodes = require("../response_codes");
+const nodeuuid = require("uuid").v1;
+const farmhash = require("farmhash");
+const AlluxioClient = require("../models/alluxioClient");
+const slash = require("slash");
+
+const generateFoldernames = (fileName, dirLevels) => {
```

Comment:
Is this identical to the one we use in fs? if so we may want to move this out to a common file (in utils or something)